rfhub2/cli/population.py

```diff
@@ -0,0 +1,175 @@
+# from pathlib import Path
+import robot.libraries
+from robot.errors import DataError
+from robot.libdocpkg import LibraryDocumentation
+from requests import session, post
+from rfhub2.config import APP_INTERFACE, APP_PORT, BASIC_AUTH_USER, BASIC_AUTH_PASSWORD
+import os
+import re
+
+RESOURCE_PATTERNS = {".robot", ".txt", ".tsv", ".resource"}
+ALL_PATTERNS = (RESOURCE_PATTERNS | {".xml", ".py"})
+PROTOCOL = 'http://'
+API_V1 = 'api/v1'
+
+
+class LibraryPopulation(object):
+
+    def __init__(self, paths, no_installed_keywords):
+        self.paths = paths
+        self.auth = (BASIC_AUTH_USER, BASIC_AUTH_PASSWORD)
+        self.no_installed_keywords = no_installed_keywords
+
+    def add_collections(self) -> None:
+        """
+        Traverses through paths and adds libraries to rfhub.
+        :return:
+        """
+
+        def traverse_paths(path):
```

Comment:
I think the function does not need to be nested, you can add `_` to indicate that it is private

Reply:
Extracted and unnested. Valid point with tests